corehq/util/test_utils.py

```diff
@@ -169,6 +170,40 @@ class flag_disabled(flag_enabled):
     enabled = False
 
 
+@contextmanager
+def disable_overpatch(patch_obj):
+    """Prevent class decorator from masking method decorator"""
+    def apply_to(patch):
+        def decorate_callable(func):
+            key = getkey(patch)
+            if any(key == getkey(p) for p in getattr(func, "patchings", [])):
+                # do not decorate if already patched for this oneshot
+                return func
+            return real_decorate_callable(func)
+
+        def copy():
+            return apply_to(real_copy())
+
+        real_decorate_callable = patch.decorate_callable
+        real_copy = patch.copy
+        patch.decorate_callable = decorate_callable
+        patch.copy = copy
+        patches.append(patch)
+        return patch
+
+    def getkey(patch):
+        return patch.getter(), patch.attribute
```

Comment:
This looks like it only identifies the thing being patched, not the patch itself, is that right?  Does that mean it'd break if the method had something unrelated patched?  Should this special type of patch have a unique identifier that's referenced here?  For example, I'd expect these things to continue to work:

* Using `@flag_enabled` on the class while `@patch`ing some other random method on a method
* Using `@flag_enabled` on the class and another `@flag_enabled` on a different feature flag at the method level (or also at the class level)

Just added a suggestion for an additional test demonstrating what I mean

Reply:
> This looks like it only identifies the thing being patched, not the patch itself, is that right?

Correct. It identifies the specific attribute and the object on which it is being patched.

> Does that mean it'd break if the method had something unrelated patched?

I don't think so...

> * Using `@flag_enabled` on the class while `@patch`ing some other random method on a method

This should be fine since the patch of some other random thing would reference different attribute.

> * Using `@flag_enabled` on the class and another `@flag_enabled` on a different feature flag at the method level (or also at the class level)

In this case the object being patched (a toggle object) would be different for each `@flag_enabled`.

Reply:
Ah okay I think I get it now - didn't realize that `patch.getter` would return the specific object, but I played around with this locally and I see it now